### PR TITLE
Upgrade jest to 20.1.0-delta.1

### PR DESCRIPTION
**Pull request by sebmarkbage**
This version includes multi-project support. This support is a bit broken in 20.0.0 so I use a pre-release. This PR doesn't yet enable it though. It's just a plain upgrade.

This required some minor upgrades to our uses of jasmine APIs and record-tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,11 +64,11 @@
     "gzip-size": "^3.0.0",
     "http-server": "^0.9.0",
     "http2": "^3.3.6",
-    "jest": "^19.0.1",
-    "jest-config": "^19.0.1",
-    "jest-jasmine2": "^19.0.1",
-    "jest-matchers": "^19.0.0",
-    "jest-runtime": "^19.0.1",
+    "jest": "20.1.0-delta.1",
```
Comment (gaearon) [inline package.json]: Interesting release naming.

File: scripts/jest/test-framework-setup.js [hunk 2]
```diff
@@ -13,6 +13,10 @@ var newError = function() {
 
 console.error = newError;
 
+// TODO: Stop using spyOn in all the test since that seem deprecated.
```
Comment (gaearon) [inline scripts/jest/test-framework-setup.js]: What is suggested alternative?

File: package.json (hunk 1 above)
Comment (nhunzaker) [inline package.json]: Was there an alpha, beta, and gamma? 🐱

File: scripts/jest/test-framework-setup.js (hunk 2 above)
Comment (sebmarkbage) [inline scripts/jest/test-framework-setup.js]: I'm not sure. I think it's probably `jest.mock`, maybe?

**Comment by gaearon** [comment]
Looks like this breaks watcher. https://github.com/facebook/jest/issues/4011

File: package.json (hunk 1 above)
Comment (thymikee) [inline package.json]: @nhunzaker There was alpha, beta, and chi ;)

File: scripts/jest/test-framework-setup.js (hunk 2 above)
Comment (thymikee) [inline scripts/jest/test-framework-setup.js]: [`jest.spyOn`](http://facebook.github.io/jest/docs/jest-object.html#jestspyonobject-methodname)

**Pull request merged**